From 1f95e5c0fe2b34acd59a766c5ec0da4e0fa8e93c Mon Sep 17 00:00:00 2001
From: willies952002 <admin@domnian.com>
Date: Mon, 25 Dec 2017 21:26:15 -0500
Subject: [PATCH] Entity Serialization

Add Method to Spawn an Entity at a Location if it's not in any world (invalid)

Manually Applied Patch from Empire Minecraft - Originally by Aikar
---
 src/main/java/com/domnian/paperdragon/api/DAPI_Entity.java | 4 ++++
 src/main/java/org/bukkit/entity/Entity.java                | 5 +++++
 2 files changed, 9 insertions(+)

diff --git a/src/main/java/com/domnian/paperdragon/api/DAPI_Entity.java b/src/main/java/com/domnian/paperdragon/api/DAPI_Entity.java
index 493aec97..0664cabf 100644
--- a/src/main/java/com/domnian/paperdragon/api/DAPI_Entity.java
+++ b/src/main/java/com/domnian/paperdragon/api/DAPI_Entity.java
@@ -1,10 +1,14 @@
 package com.domnian.paperdragon.api;
 
 import org.bukkit.Location;
+import org.bukkit.World;
 import org.bukkit.entity.*;
 
 public interface DAPI_Entity {
 
+    byte[] serializeEntity(Entity craftentity);
+    Entity deserializeEntity(byte[] data, World world);
+
     boolean isEntityDisabled(org.bukkit.entity.Entity entity);
     void setItemCanDespawn(Item item, boolean canDespawn);
 
diff --git a/src/main/java/org/bukkit/entity/Entity.java b/src/main/java/org/bukkit/entity/Entity.java
index 69c1db4a..2fd158b6 100644
--- a/src/main/java/org/bukkit/entity/Entity.java
+++ b/src/main/java/org/bukkit/entity/Entity.java
@@ -21,6 +21,11 @@ import org.bukkit.event.player.PlayerTeleportEvent.TeleportCause;
  */
 public interface Entity extends Metadatable, CommandSender, Nameable {
 
+    // PaperDragon start
+    public boolean spawnAt(Location loc);
+    public boolean spawnAt(Location loc, org.bukkit.event.entity.CreatureSpawnEvent.SpawnReason spawnReason);
+    // PaperDragon end
+
     /**
      * Gets the entity's current position
      *
-- 
2.16.2

